samples/cpp/tutorial_code/Histograms_Matching/MatchTemplate_Demo.cpp

```diff
@@ -31,15 +32,20 @@ int main( int argc, char** argv )
   if (argc < 3)
   {
     cout << "Not enough parameters" << endl;
-    cout << "Usage:\n./MatchTemplate_Demo <image_name> <template_name>" << endl;
+    cout << "Usage:\n./MatchTemplate_Demo <image_name> <template_name> [<mask_name>]" << endl;
     return -1;
   }
 
   /// Load image and template
   img = imread( argv[1], IMREAD_COLOR );
   templ = imread( argv[2], IMREAD_COLOR );
 
-  if(img.empty() || templ.empty())
+  if(argc > 3) {
+    use_mask = true;
+    mask = imread(argv[3], IMREAD_COLOR);
```

Comment:
it's stated that "The mask should be a greyscale image", but here it's forced to 3 channels